pkg/aws/awscontrolplane/validate_awscontrolplane_test.go

```diff
@@ -12,6 +12,131 @@ import (
 	"github.com/giantswarm/aws-admission-controller/pkg/unittest"
 )
 
+func TestAZReplicaMatch(t *testing.T) {
+	testCases := []struct {
+		ctx  context.Context
+		name string
+
+		allowed  bool
+		replicas int
+	}{
+		{
+			ctx:  context.Background(),
+			name: "case 0",
+
+			allowed:  true,
+			replicas: 1,
+		},
+		{
+			ctx:  context.Background(),
+			name: "case 1",
+
+			allowed:  false,
+			replicas: 3,
+		},
+	}
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			var err error
+
+			// Create a new logger that is used by all admitters.
+			var newLogger micrologger.Logger
+			{
+				newLogger, err = micrologger.New(micrologger.Config{})
+				if err != nil {
+					panic(microerror.JSON(err))
+				}
+			}
+
+			fakeK8sClient := unittest.FakeK8sClient()
+			validate := &Validator{
+				k8sClient: fakeK8sClient,
+				logger:    newLogger,
```

Comment:
I think you can drop the logger if it is not relevant for the test itself. 

```suggestion
				logger:    microloggertest.New(),
```

Reply:
I'm not familiar with `microloggertest` but I'll try and see if this works.

Reply:
It is just some fake implementation and acts as drop in replacement for unit tests.